pytest_twisted.py

```diff
@@ -275,10 +288,27 @@ def init_qt5_reactor():
 def init_asyncio_reactor():
     from twisted.internet import asyncioreactor
 
-    _install_reactor(
-        reactor_installer=asyncioreactor.install,
-        reactor_type=asyncioreactor.AsyncioSelectorReactor,
-    )
+    try:
+        _install_reactor(
+            reactor_installer=asyncioreactor.install,
+            reactor_type=asyncioreactor.AsyncioSelectorReactor,
+        )
+    except NotImplementedError as e:
+        import asyncio
+
+        _, _, traceback_object = sys.exc_info()
+        stack_summary = traceback.extract_tb(traceback_object)
+        source_function_name = stack_summary[-1].name
+        event_loop = asyncio.get_event_loop()
```

Comment:
I'm -1 on this. I realize you're trying to scope it to this specific case to avoid a lying error message, but IMO this relies way too heavily on implementation details.

I think preserving the traceback is enough for context, and then just change the error to say what we *think* happened... Phrased like "If you're on windows and python 3.8+, you might need to..." perhaps.

Reply:
Perhaps my head was still in the 'identify the issue and resolve it' mode where you want to be sure.  Given this is just going to refine the error message which can be worded as a guess it is more acceptable to... guess.

Maybe instead of more granular `NotImplementedError`d raisers should include the function object raising it for identification.

Plus I suppose back to 'this is really twisted's problem to address' justifying less effort.